import_tracker/rest.py

```diff
@@ -1,22 +1,69 @@
 # -*- coding: utf-8 -*-
 from girder.api import access
+from girder.utility import path, model_importer
 from girder.api.describe import Description, autoDescribeRoute
 from girder.constants import SortDir
+from girder.api.rest import boundHandler
+from girder.models.assetstore import Assetstore
+
+
 from .models import AssetstoreImport
 
 from bson.objectid import ObjectId
 
 
+def processCursor(cursor, user):
+    lookedupAssetstores = {}
+    results = list(cursor)
+
+    for row in results:
+        if row['assetstoreId'] not in lookedupAssetstores:
+            assetstore = list(Assetstore().find({'_id': row['assetstoreId']}))
+            lookedupAssetstores[row['assetstoreId']] = assetstore[0]['name']
+
+        row['_assetstoreName'] = lookedupAssetstores[row['assetstoreId']]
+        model = model_importer.ModelImporter.model(row['params']['destinationType'])
+        doc = model.load(row['params']['destinationId'], user=user)
+        row['_destinationPath'] = path.getResourcePath(
+            row['params']['destinationType'],
+            doc
+            )
```

Comment:
You need to add `user=user` to the `getResourecPath` call.

Reply:
tool tips added in aa816989b9f996841f798cca742548205fe97b9a

guard added in 66dcff33a5c5411165f4195a4de8ad376bd12ba9

The import will actually fail if the destination no longer exists (so will looking up the destination path). In this case I have changed the destination path to 'does not exist' and in 61a2015780d9371ee7e384a193c906e1674b5423 I disable the re-import button